src/main/java/com/internetitem/logback/elasticsearch/writer/ElasticsearchWriter.java

```diff
@@ -32,25 +34,45 @@ public ElasticsearchWriter(ErrorReporter errorReporter, Settings settings, HttpR
 			: Collections.<HttpRequestHeader>emptyList();
 
 		this.sendBuffer = new StringBuilder();
+		this.backupBuffer = new StringBuilder();
 	}
 
 	public void write(char[] cbuf, int off, int len) {
+		if (!bufferExceeded && backupBuffer.length() > 0) {
```

Comment:
This was rewritten completely.

The previous implementation over-written the backupBuffer and only checked size after that.
Now we check if the current chunk fits, and only append it if it does, this will ensure the request size will be within the hard bounds set by AWS.

If it doesn't fit, we switch to the backupBuffer, and append to that, unless that's full as well, in which case we throw away the log (what else could we do, have to bound it somewhere)